### PR TITLE
tmpi: init at 1.0.0

### PR DESCRIPTION
**Pull request by Vasissualiyp**
## Description of changes

Added [tmpi](https://github.com/Azrael3000/tmpi?tab=readme-ov-file) - a shell script that is useful when running multi-core jobs with mpirun for debugging purposes. Uses tmux to look into each process separately. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/tm/tmpi/package.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  propagatedBuildInputs = [ mpi mpich reptyr tmux ];
+
+  buildInputs = [ autoconf ];
+
+  installPhase = ''
+    runHook preInstall
+
+    # Create a wrapper for the script to set the correct environment variables
+    mkdir -p $out/bin
+    cat > $out/bin/tmpi <<EOF
```
Comment (mattpolzin) [inline pkgs/by-name/tm/tmpi/package.nix:31]: curious if `wrapProgram` would work for you here.

**Review by mattpolzin — COMMENTED** [review]
I don't suppose either of my comments are blockers for this package, but I noticed a couple of things you could do differently.

**Comment by Vasissualiyp** [comment]
Force-pushed all requested changes, namely the version name adjustment and makeWrapper addition.

**Comment by mattpolzin** [comment]
Looks good. I can test this in a bit. I did try testing on my `x86_64-darwin` machine first and noticed that `reptyr` does not support that platform -- might be worth inheriting `reptyr`'s `platforms` (assuming that's the most restrictive dependency of this derivation). If you do make that change, you could tweak the commit message since the version is no longer `1.0.0`.

At any rate, I'll be at my linux laptop soon to make sure it builds for me so I can give it a 👍.

**Review by mattpolzin — APPROVED** [review]
Built and tested.

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 323875` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tmpi</li>
  </ul>
</details>

**Pull request merged**